backend/contest/room.go

```diff
@@ -169,6 +170,18 @@ func (r *roomImpl) Join(name string) Player {
 	return player
 }
 
+func (r *roomImpl) Leave(playerKey string) Player {
+	player := r.players[playerKey]
+	r.notifyPlayers(
+		func(p Player) {
+			p.NotifyPlayerLeft(player.Name, player.Key)
+		},
+	)
+	delete(r.players, playerKey)
```

Comment:
You could add 

```
if len(r.players) == 0 {
		r.finished = true
}
```


This should delete the room within 5 seconds (the delete room event is logged to cmd).